node/src/main/scala/org/bitcoins/node/PeerFinder.scala

```diff
@@ -166,19 +172,21 @@ case class PeerFinder(
     //delete try queue
     _peersToTry.clear()
 
-    _peerData.foreach(_._2.client.close())
+    val closeF = Future.sequence(_peerData.map(_._2.client.map(_.close())))
 
-    AsyncUtil
+    val waitStopF = AsyncUtil
       .retryUntilSatisfied(_peerData.isEmpty,
```

Comment:
Why wait for `_peerData.isEmpty`, is it not good enough for `closeF` future to be completed?

Reply:
So `peerData` map is such that the keys correspond to a running actor. The `close` called here has a return type of `Unit` and just sends a message to the client to stop the actor along with properly changing the state of `PeerMessageReceiver`. The actor actually stops sometime in the future, and the key-value pair is deleted from the map in the `postStop` callback of actors, so this is just an additional check to ensure that all actors have actually stopped.